### DISCUSSION TITLE
Only interested in Windows

### DISCUSSION THREAD

**Discussion by mjanulaitis** [Live-Charts/LiveCharts2#825] [General]
How can we remove SkiaSharp and instead use native WPF drawing?

**Comment by beto-rodriguez** [comment]
Can I get some feedback? what is wrong with SkiaSharp?

**Reply by mjanulaitis** [reply]
It looked at the binary directory so I could see what I need to add to my WIX installer and noticed the top 5 dlls and dylib (whatever those are) are a total are:  libSkiaSharp.dylib (15 MBs), libSkiaSharp.dll (7 MBs), libHarfBuzzSharp.dylib (2 MBs), libHarfBuzzSharp.dll (< 1 MB) and SkiaSharp.dll (< 1 MB) for a total of about 26 MBs.  I provide my customers with both exe and msi installers so that adds 52 MBs to my upload not to mention there are a few other dlls like HarfBuzzSharp.dll, LiveChartsCore.SkiaSharpView.dll, LiveChartsCore.SkiaSharpView.WPF.dll, LiveChartsCore.SkiaSharpView.WPF.xml, SkiaSharp.Views.Desktop.Common.dll, SkiaSharp.Views.WPF.dll and SkiaSharp.HarfBuzz.dll which it appears all need to be shipped. That's a lot of dlls which are just all wrappers around Windows GDI (or possibly another API???).

**Reply by mjanulaitis** [reply]
I also just noticed in my x64 directory another 10 MBs of files :(.  libSkiaSharp.dll and libHarfBuzzSharp.dll.

**Reply by beto-rodriguez** [reply]
This is a good point; I will explore the SkiaSharp repo to find if there is something we can do here.

On the other hands using SkiaSharp has also a lot of benefits (from the library perspective), the library can now run in the [browser](https://blazor-livecharts.controli.app/), mobile, Linux and MacOS.

Also, in theory we could consume the core of the library and draw everything using something else, but even the code is designed for that, there are a few things that needs to be cleaned, maybe in a future this will be possible.

**Reply by mjanulaitis** [reply]
It seems the skia is adding files to the binary directory not needed.  Specifically the dylib appear to be Mac libraries.  I deleted them and the sample worked.

**Comment by mjanulaitis** [comment]
When comparing memory footprints, the sample app is taking 100 MBs of memory with the basic pie loaded.  My app, which is massive, consumes ~75 MBs of memory loading a PieChart using LiveCharts 1.  So it seems the skia libraries are consuming way more memory then necessary.  This is a really bummer.  I want to move forward with the new version because the ChartUpdater is no longer used which should significantly improve performance IMO however the memory footprint is just too much.

**Reply by beto-rodriguez** [reply]
There is a huge difference in performance in v2, take a look at this article that explains more about the future performance of the library:

https://github.com/beto-rodriguez/LiveCharts2/releases/tag/v2.0.0-beta.350

There are multiple improvements in v2, about the memory used, I will take a look on that, beta-700 version improved memory allocation.
